src/cantools/database/can/database.py

```diff
@@ -146,86 +146,124 @@ def is_similar(self,
+                add_diff(path_parts, type(a), type(b), 'type-mismatch')
+                return
+
+            if a is None:
+                return
+
+            if isinstance(a, (int, str, set, bool)):
+                if a != b:
+                    add_diff(path_parts, a, b, 'value-mismatch')
+                return
+
+            if isinstance(a, float):
+                if abs(a) > 1:
+                    if abs(1.0 - b / a) > tolerance:
+                        add_diff(path_parts, a, b, 'float-rel-diff')
+                else:
+                    if abs(b - a) > tolerance:
+                        add_diff(path_parts, a, b, 'float-abs-diff')
+                return
+
+            if isinstance(a, (list, tuple)):
+                if len(a) != len(b):
+                    add_diff(path_parts, len(a), len(b), 'length-mismatch')
+                min_len = min(len(a), len(b))
+                for i in range(min_len):
+                    compare(a[i], b[i], [*path_parts, f'[{i}]'])
+                return
+
+            if isinstance(a, (dict, OrderedDict)):
+                keys_a = set(a.keys())
+                keys_b = set(b.keys())
+                for k in sorted(keys_a - keys_b, key=lambda x: str(x)):
+                    add_diff([*path_parts, f'[{k!r}]'], a[k], None, 'key-only-in-a')
+                for k in sorted(keys_b - keys_a, key=lambda x: str(x)):
+                    add_diff([*path_parts, f'[{k!r}]'], None, b[k], 'key-only-in-b')
+                for k in sorted(keys_a & keys_b, key=lambda x: str(x)):
+                    compare(a[k], b[k], [*path_parts, f'[{k!r}]'])
+                return
+
+            # get attributes
+            a_names = dir(a)
+            b_names = dir(b)
+
+            if not include_format_specifics:
+                for x in ('dbc', 'autosar'):
+                    if x in a_names:
+                        a_names.remove(x)
+                    if x in b_names:
+                        b_names.remove(x)
+
+            if a_names != b_names:
+                add_diff(path_parts, sorted(a_names), sorted(b_names), 'attrib-names-mismatch')
+                return
+
+            for name in a_names:
+                if name == 'messages' and hasattr(a, '_frame_id_to_message'):
+                    # compare messages independent of order
+                    compare(a._frame_id_to_message, b._frame_id_to_message, ["_frame_id_to_message"])
+                    continue
+
+                if name == 'signals' and hasattr(a, '_signal_dict'):
+                    # compare messages independent of order
+                    compare(a._signal_dict, b._signal_dict, ["_signal_dict"])
+                    continue
+
+                if name.startswith('_'):
+                    # skip private attributes
+                    continue
+
+                a_attr = getattr(a, name)
+                b_attr = getattr(b, name)
+                if callable(a_attr) or callable(b_attr):
```

Comment:
I suppose that if one of these attributes is callable, but the other is not, a `diff` should be added.